File: 09) Java Advanced - January 2019/14. Defining Classes - Exercises/05. Car Salesman/Car/Parts/Engine.java
package Car.Parts;

public class Engine {

    private String model;

    private int power;

    private int displacement;

    private String efficiency;

    public Engine(String model, int power) {
        setModel(model);
        setPower(power);
    }

    public Engine(String model, int power, int displacement) {
        this(model, power);
        setDisplacement(displacement);
    }

    public Engine(String model, int power, int displacement, String efficiency) {
        this(model, power, displacement);
        setEfficiency(efficiency);
    }



    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public int getPower() {
        return power;
    }

    public void setPower(int power) {
        this.power = power;
    }

    public int getDisplacement() {
        return displacement;
    }

    public void setDisplacement(int displacement) {
        this.displacement = displacement;
    }

    public String getEfficiency() {
        return efficiency;
    }

    public void setEfficiency(String efficiency) {
        this.efficiency = efficiency;
    }
}

File: Software University/04) PHP Web Development Basics/10) OOP Basics - Exercises/07. Car Salesman/Engine.php
<?php

namespace Van\Classes;

class Engine
{
    /**
     * @var string
     */
    private $model;

    /**
     * @var int
     */
    private $power;

    /**
     * @var int
     */
    private $displacement;

    /**
     * @var string
     */
    private $efficiency;

    /**
     * Engine constructor.
     *
     * @param string $model
     * @param int $power
     */
    public function __construct(string $model, int $power)
    {
        $this->model = $model;
        $this->power = $power;
    }

    /**
     * Checks if the optional property Displacement is set
     *
     * @return bool
     */
    public function hasDisplacement(): bool
    {
        return isset($this->displacement);
    }

    /**
     * Checks if the optional property Efficiency is set
     *
     * @return bool
     */
    public function hasEfficiency(): bool
    {
        return isset($this->efficiency);
    }

    /**
     * @return string
     */
    public function getModel(): string
    {
        return $this->model;
    }

    /**
     * @param string $model
     */
    public function setModel(string $model)
    {
        $this->model = $model;
    }

    /**
     * @return int
     */
    public function getPower(): int
    {
        return $this->power;
    }

    /**
     * @param int $power
     */
    public function setPower(int $power)
    {
        $this->power = $power;
    }

    /**
     * @return int
     */
    public function getDisplacement(): int
    {
        return $this->displacement;
    }

    /**
     * @param int $displacement
     */
    public function setDisplacement(int $displacement)
    {
        $this->displacement = $displacement;
    }

    /**
     * @return string
     */
    public function getEfficiency(): string
    {
        return $this->efficiency;
    }

    /**
     * @param string $efficiency
     */
    public function setEfficiency(string $efficiency)
    {
        $this->efficiency = $efficiency;
    }
}
